langlib/lang.array/src/main/java/org/ballerinalang/langlib/array/utils/ArrayUtils.java

```diff
@@ -81,6 +89,38 @@ public static BError createOpNotSupportedError(Type type, String op) {
         return ErrorCreator.createError(getModulePrefixedReason(ARRAY_LANG_LIB,
                                                                 OPERATION_NOT_SUPPORTED_IDENTIFIER),
                                         StringUtils.fromString(format("%s not supported on type '%s'", op,
-                                                                       type.getQualifiedName())));
+                                                                      type.getQualifiedName())));
+    }
+
+    public static BArray createEmptyArrayFromTuple(BArray arr) {
+        Type arrType = arr.getType();
+        TupleType tupleType = (TupleType) arrType;
+        List<Type> memTypes = new ArrayList<>();
+        List<Type> tupleTypes = tupleType.getTupleTypes();
+        boolean isSameType = true;
+        Type sameType = null;
+        if (!tupleTypes.isEmpty()) {
+            sameType = tupleTypes.get(0);
+            memTypes.add(sameType);
+        }
+        for (int i = 1; i < tupleTypes.size(); i++) {
+            isSameType &= sameType == tupleTypes.get(i);
+            memTypes.add(tupleTypes.get(i));
+        }
+        Type restType = tupleType.getRestType();
+        if (restType != null) {
+            isSameType = false;
+            memTypes.add(restType);
+        }
+        if (isSameType) {
+            ArrayType type = TypeCreator.createArrayType(sameType);
```

Comment:
Should the new array type always be a union type? AFAIU we check duplicate types? Can we simply have a set for memTypes?

Reply:
Shall we add a comment here?

Reply:
According to James it has to be union type: https://github.com/ballerina-platform/ballerina-spec/issues/731#issuecomment-774936212

Reply:
Added a comment in the code